breakglass.go

```diff
@@ -81,6 +82,64 @@ func loadAuthorizedKeys(path string) (map[string]bool, error) {
 	return result, nil
 }
 
+func getEC2MetadataToken() (string, error) {
+	req, _ := http.NewRequest("PUT", "http://169.254.169.254/latest/api/token", nil)
+	req.Header.Add("X-aws-ec2-metadata-token-ttl-seconds", "21600")
+	res, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return "", fmt.Errorf("failed to get metadata token: %w", err)
+	}
+	defer res.Body.Close()
+	if res.StatusCode != 200 {
+		return "", fmt.Errorf("failed to get metadata token: %v", res.Status)
+	}
+	all, err := io.ReadAll(res.Body)
+	if err != nil {
+		return "", fmt.Errorf("failed to read metadata token: %w", err)
+	}
+	return strings.TrimSpace(string(all)), nil
+}
+
+func loadAWSEC2SSHKeys() ([]byte, error) {
+	token, err := getEC2MetadataToken()
+	if err != nil {
+		return nil, err
+	}
+	var buf bytes.Buffer
+	getKeyIndex := func(idx int) error {
+		req, _ := http.NewRequest("GET", fmt.Sprintf("http://169.254.169.254/latest/meta-data/public-keys/%d/openssh-key", idx), nil)
+		req.Header.Add("X-aws-ec2-metadata-token", token)
+		res, err := http.DefaultClient.Do(req)
+		if err != nil {
+			return err
+		}
+		defer res.Body.Close()
+		if res.StatusCode != 200 {
+			return errors.New(res.Status)
+		}
+		all, err := io.ReadAll(res.Body)
+		if err != nil {
+			return err
+		}
+		buf.Write(all)
+		if !bytes.HasSuffix(buf.Bytes(), []byte("\n")) {
```

Comment:
Does the SSH package need this trailing newline? Can you document that in a comment if so, or remove if not strictly needed?

Reply:
It's needed as a delimiter between multiple SSH keys if you've told AWS that you have more than 1. It's paranoia because I can't find exact specs on whether they include one or not, or if it depends on how the user configures it.

I'll leave a comment.

Reply:
Done.